### PR TITLE
Dev/ch6941/updater la question du sexe sur le formulaire

### PR DESCRIPTION
**Pull request by Raulnet**

### CODE REVIEW

**Comment by Raulnet** [comment]
no effect in the admin board. the field gender was not used for filter or Query. but for now the customGender could'not be edited in the Admin board and on the profile

File: src/Entity/Adherent.php [hunk 1]
```diff
@@ -611,6 +620,11 @@ public function getGender(): ?string
         return $this->gender;
     }
 
+    public function getCustomGender(): ?string
```
Comment (ottaviano) [inline src/Entity/Adherent.php]: what about a setter?

File: templates/adherent/profile.html.twig [hunk 2]
```diff
@@ -42,6 +43,11 @@
             {{ form_widget(form.gender, { attr: { class: 'register__form__gender' } }) }}
             {{ form_errors(form.gender) }}
         </div>
+        <div class="l__col l__col--top register__form--mid visually-hidden">
+            {{ form_label(form.customGender, 'Personnalisez votre genre', { 'label_attr': {'class': 'text--gray'} }) }}
```
Comment (ottaviano) [inline templates/adherent/profile.html.twig]: `label_attr: { class: 'text--gray' }`

File: templates/facebook/index.html.twig [hunk 3]
```diff
@@ -24,7 +24,7 @@
                         <label for="mentions_legales">
                             En cochant cette case, vous autorisez En Marche à utiliser les informations suivantes pour
                             activer un filtre La République En Marche sur votre compte Facebook (photo de profil Facebook, nom, prénom,
-                            âge, sexe et adresse mail).
+                            âge, genre et adresse mail).
```
Comment (ottaviano) [inline templates/facebook/index.html.twig]: are you sure about this change?

File: src/Controller/EnMarche/MembershipController.php [hunk 4]
```diff
@@ -94,7 +94,6 @@ public function adhesionAction(Request $request, GeoCoder $geoCoder): Response
             ->add('submit', SubmitType::class, ['label' => 'Je rejoins La République En Marche'])
             ->handleRequest($request)
         ;
-
```
[the remark this answers is not in the document]
Comment (Raulnet) [reply src/Controller/EnMarche/MembershipController.php]: restored.

File: src/Entity/Adherent.php [hunk 5]
```diff
@@ -118,6 +118,13 @@ class Adherent implements UserInterface, UserEntityInterface, GeoPointInterface,
      */
     private $gender;
 
+    /**
+     * @ORM\Column(length=80, nullable=true)
+     *
+     * @JMS\Groups({"adherent_change_diff"})
```
[the remark this answers is not in the document]
Comment (Raulnet) [reply src/Entity/Adherent.php]: removed.

File: src/Entity/Adherent.php (hunk 1 above)
Comment (Raulnet) [reply src/Entity/Adherent.php]: Never used the propertie is set by DTO, there are few setter in class Adherent

File: templates/adherent/profile.html.twig (hunk 2 above)
Comment (Raulnet) [reply templates/adherent/profile.html.twig]: fixed.

File: templates/facebook/index.html.twig (hunk 3 above)
Comment (Raulnet) [reply templates/facebook/index.html.twig]: restored

File: src/Membership/MembershipRequest.php [hunk 6]
```diff
@@ -7,28 +7,29 @@
...
  * @AssertUniqueMembership(groups={"Registration", "Update"})
+ * @AssertCustomGender(groups={"Registration", "Update"})
  */
 class MembershipRequest implements MembershipInterface
 {
     /**
      * @var string|null
      *
      * @Assert\NotBlank(message="common.gender.not_blank", groups={"Update"})
-     * @Assert\Choice(
```
[the remark this answers is not in the document]
Comment (Raulnet) [reply src/Membership/MembershipRequest.php]: restored

File: src/Entity/Adherent.php [hunk 7]
```diff
@@ -393,7 +400,8 @@ public static function create(
         string $registeredAt = 'now',
         ?array $tags = [],
         ?array $referentTags = [],
-        ?array $mandates = null
+        ?array $mandates = null,
```
[the remark this answers is not in the document]
Comment (Raulnet) [reply src/Entity/Adherent.php]: edited

**Pull request merged**